sql/mysqld.cc

```diff
@@ -2081,39 +2081,29 @@ static void clean_up(bool print_message)
 
 #ifndef EMBEDDED_LIBRARY
 
-/**
-  This is mainly needed when running with purify, but it's still nice to
-  know that all child threads have died when mysqld exits.
-*/
 static void wait_for_signal_thread_to_end()
 {
-  uint i, n_waits= DBUG_EVALUATE("force_sighup_processing_timeout", 5, 100);
```

Comment:
you removed "force_sighup_processing_timeout", but it's used in the test

Reply:
I removed that part of the test, too, because as far as I can see it wasn't necessary; it was only needed because we had to wait, but we had to wait because of the bug that was in the `setup_hand` function.

Reply:
I should add that by using `pthread_join`, we'll wait for all `SIGHUP` processing to finish, so there's no need for a special timeout.  Joining pthreads is considered a best practice for the pthread library (unless the thread is created in a detached state, which isn't the case here).

Reply:
`rpl.rpl_shutdown_sighup` continues to succeed with these changes, because the `SIGHUP` processing concludes as expected, without the unnecessary extra waiting semantics.  Those were brittle because `sleep`s (and equivalent) are not reliable for multithreaded coordination.